src/Components/PaperEditor/TranscriptsContainer/TranscriptTabPane/TranscriptTabContent/index.js

```diff
@@ -424,38 +424,33 @@ const TranscriptTabContent = (props) => {
     });
   };
 
-  const onLabelCreate = (newLabel) => {
-    createLabel(newLabel);
-    const tempLabels = labels;
-    tempLabels.push(newLabel);
-    setLabels(tempLabels);
-    props.trackEvent({ category: 'paperEditor transcriptsTab', action: 'label create ' });
-  };
-
   const onLabelUpdate = (labelId, updatedLabel) => {
-    setLabels(() => [ ...tempLabels, updatedLabel ]);
+    setLabels(() => [ ...labels, updatedLabel ]);
     LabelsCollection.putItem(labelId, updatedLabel);
     props.trackEvent({ category: 'paperEditor transcriptsTab', action: `label update ${ labelId } ${ updatedLabel }` });
   };
 
-  const onLabelDelete = (labelId) => {
+  const onLabelDelete = (labelToDelete) => {
```

Comment:
One thing I noticed while testing is that if you delete a label from the menu, the parts in the transcript that have that label automatically get assigned a new label colour, rather than reverting to not having a label. I know changing / deleting actual selections is a later ticket, so idk if this falls under that?

Reply:
yeah this was happening before and we talked about it a bit yesterday in the clean up meeting and my understanding is that this will be dealt with in the deleting label ticket - in fact I'm going to go and add it to that ticket right now!